build/commands/lib/util.js

```diff
@@ -611,7 +612,11 @@ const util = {
   buildTargets: async (
     targets = config.buildTargets,
     options = config.defaultOptions,
+    noGnGen = false
   ) => {
+    if (!noGnGen) {
+      await util.generateNinjaFiles()
```

Comment:
this was needed because CI switches from component -> static after the build and there is no create_dist target for ios to generate the output directory here

Reply:
also doing this here will most likely break the release builds, because the `test` command is missing a lot of stuff that's passed in release build jobs to `create_dist/build`. You need to be careful here.

Reply:
I added this specifically because it broke on CI. It builds debug for the main builds and static for tests. On other platforms it does create_dist as static as well so it does gn gen, but there is no create_dist for ios. @wknapik is this going to cause a problem? If it is, we could just run a normal static build for ios prior to test instead of `create_dist`

Reply:
> If we switch to building Static in iOS PRs, as you suggest, do we still want to do a Debug build at the end?

A question to @brave/ios team. Another question: is there a reason to build `Debug` in a release pipeline?

Reply:
> Should we pass --target_environment=device in build in PRs?

I don't think we need this in particular, we only build device for public builds because thats what we upload to the App Store

> You're suggesting building Debug -> Static, but in public builds we already do Static -> Debug. Does the order matter?

Debug builds are specifically for the `fastlane ios build_without_testing` and `fastlane ios test_without_building` calls on the CI side, so the order only matters as long as the debug build is made before calling `build_without_testing`

Reply:
I reverted the change, but we still need to figure out what to do here. I think for this PR we should just add `build Static` for ios and any changes to `Debug` build should be done separately @wknapik

Reply:
> Should gn-check in PRs vs public builds use different values?

no args need to be passed to gn check at all for any build except for the output directory

> I'll take this to mean that we should pass --target_environment=simulator

`--target_environment=simulator` is necessary for whatever build we use for tests. Are we currently even building anything other than Debug without `ios_output_xml` for ios PR builder? If we're doing fastlane tests with debug build then maybe the simplest thing here is just to change test from Static -> Debug

Reply:
also we can't change any args from build -> test or it won't work correctly because we don't run gn gen for tests so we have to build and run tests with `--target_environment=simulator`

Reply:
CI changed from Static -> Debug for now to resolve this for now